javascript-carousel/main.js

```diff
@@ -0,0 +1,312 @@
+var $displayImage = document.querySelector('.display-image');
+var $circleZero = document.querySelector('.zero');
+var $circleOne = document.querySelector('.one');
+var $circleTwo = document.querySelector('.two');
+var $circleThree = document.querySelector('.three');
+var $circleFour = document.querySelector('.four');
+var $circleContainer = document.querySelector('.circles');
+var intervalCounter = 0;
+var pictureArray = [
+  'images/001.png',
+  'images/004.png',
+  'images/007.png',
+  'images/025.png',
+  'images/039.png'
+];
+
+var intervalId = setInterval(rotate, 3000);
+
+function rotate() {
+  intervalCounter++;
+  $displayImage.setAttribute('src', pictureArray[intervalCounter]);
+  clearInterval(intervalId);
+  intervalId = setInterval(rotate, 3000);
+  if (intervalCounter === 0) {
+    $circleZero.className = 'fa-solid fa-circle circle zero';
+    $circleOne.className = 'fa-regular fa-circle circle one';
+    $circleTwo.className = 'fa-regular fa-circle circle two';
+    $circleThree.className = 'fa-regular fa-circle circle three';
+    $circleFour.className = 'fa-regular fa-circle circle four';
+  } else if (intervalCounter === 1) {
+    $circleZero.className = 'fa-regular fa-circle circle zero';
+    $circleOne.className = 'fa-solid fa-circle circle one';
+    $circleTwo.className = 'fa-regular fa-circle circle two';
+    $circleThree.className = 'fa-regular fa-circle circle three';
+    $circleFour.className = 'fa-regular fa-circle circle four';
+  } else if (intervalCounter === 2) {
+    $circleZero.className = 'fa-regular fa-circle circle zero';
+    $circleOne.className = 'fa-regular fa-circle circle one';
+    $circleTwo.className = 'fa-solid fa-circle circle two';
+    $circleThree.className = 'fa-regular fa-circle circle three';
+    $circleFour.className = 'fa-regular fa-circle circle four';
+  } else if (intervalCounter === 3) {
+    $circleZero.className = 'fa-regular fa-circle circle zero';
+    $circleOne.className = 'fa-regular fa-circle circle one';
+    $circleTwo.className = 'fa-regular fa-circle circle two';
+    $circleThree.className = 'fa-solid fa-circle circle three';
+    $circleFour.className = 'fa-regular fa-circle circle four';
+  } else if (intervalCounter === 4) {
+    $circleZero.className = 'fa-regular fa-circle circle zero';
+    $circleOne.className = 'fa-regular fa-circle circle one';
+    $circleTwo.className = 'fa-regular fa-circle circle two';
+    $circleThree.className = 'fa-regular fa-circle circle three';
+    $circleFour.className = 'fa-solid fa-circle circle four';
+  } else {
+    intervalCounter = 0;
+    $circleZero.className = 'fa-solid fa-circle circle zero';
+    $circleOne.className = 'fa-regular fa-circle circle one';
+    $circleTwo.className = 'fa-regular fa-circle circle two';
+    $circleThree.className = 'fa-regular fa-circle circle three';
+    $circleFour.className = 'fa-regular fa-circle circle four';
+    $displayImage.setAttribute('src', pictureArray[intervalCounter]);
+  }
```

Comment:
This is **_very_** manual. You should find a way to do this with a loop over all the dots so that you don't need so much repetition.